packages/hardhat/contracts/strategy/CrabStrategyV2.sol

```diff
@@ -604,96 +604,89 @@ contract CrabStrategyV2 is StrategyBase, StrategyFlashSwap, ReentrancyGuard, Own
         require(offerSigner == _order.trader, "Invalid offer signature");
         require(_order.expiry >= block.timestamp, "Order has expired");
 
-        //adjust managerAmount and TraderAmount for partial fills
-        if (_remainingAmount < _order.managerAmount) {
-            _order.managerAmount = _remainingAmount;
+        //adjust quantity for partial fills
+        if (_remainingAmount < _order.quantity) {
+            _order.quantity = _remainingAmount;
         }
-        //adjust if manager is giving better price
-        _order.traderAmount = _order.managerAmount.mul(1e18).div(_clearingPrice);
-
-        IERC20(_order.traderToken).transferFrom(_order.trader, address(this), _order.traderAmount);
-
-        // if the trader is selling WETH to us i.e if we are selling oSQTH
-        if (_order.traderToken == weth) {
-            IWETH9(weth).withdraw(IWETH9(weth).balanceOf(address(this)));
-            // if last param is false, transfer happens again
-            _mintWPowerPerp(_order.trader, _order.managerAmount, _order.traderAmount, true);
-            priceAtLastHedge = _order.traderAmount.mul(1e18).div(_order.managerAmount);
+        // weth clearing price for the order
+        uint256 wethAmount = _order.quantity.mul(_clearingPrice).div(1e18);
+
+        if (_order.isBuying){
+            // trader sends weth and receives oSQTH
+            IERC20(weth).transferFrom(_order.trader, address(this), wethAmount);
+            _mintWPowerPerp(_order.trader, _order.quantity, wethAmount, true);
+            IERC20(wsqueeth).transfer(_order.trader, _order.quantity);
         } else {
-            // oSQTH in, WETH out
-            _burnWPowerPerp(_order.trader, _order.traderAmount, _order.managerAmount, true);
+            // trader sends oSQTH and receives weth
+            IERC20(wsqueeth).transferFrom(_order.trader, address(this), _order.quantity);
+            _burnWPowerPerp(_order.trader, _order.quantity, wethAmount, true);
```

Comment:
Can delete the wsqueeth transfer in above line if you, set the _burnWPowerPerp flag to false as it will be transferred in from the _order.trader

Also should probably use variable names that we have been using - wPowerPerp or oSQTH not wSqueeth

Reply:
fixed the flag - which wSqueeth var you mean or just the deleted one?